### DISCUSSION TITLE
Handling HL7 Segments

### DISCUSSION THREAD

**Discussion by jtyme** [nextgenhealthcare/connect#5683] [General]
Wondering what best practice in Mirth would be to handle possible multiple segments. We are collecting certain segments and writing the values to a external DB for storage. We have multiple formats being sent into us so trying to come up with best solution to solve multiple formats.

current code snippet is var SegOBR2 = xmlMessage['OBR']['OBR.2'][0]['OBR.2.1'].toString(); 
This normally works fine but if a client sends in something like OBR|1|006600610&N&N| this, then <OBR.2.1><OBR.2.1.1>006600610< is written to the DB and not the 00660610 value we want. 

Curious which would be the best way to handle this scenario in Mirth.

**Comment by kirbykn2** [comment]
If you don't need what's in the subcomponent, don't parse it.  This can be
configured in the channel.  The ampersand has a distinct meaning in HL7

On Thu, Feb 16, 2023 at 10:11 AM Justin Pallas ***@***.***>
wrote:

> Wondering what best practice in Mirth would be to handle possible multiple
> segments. We are collecting certain segments and writing the values to a
> external DB for storage. We have multiple formats being sent into us so
> trying to come up with best solution to solve multiple formats.
>
> current code snippet is var SegOBR2 =
> xmlMessage['OBR']['OBR.2'][0]['OBR.2.1'].toString();
> This normally works fine but if a client sends in something like
> OBR|1|006600610&N&N| this, then <OBR.2.1><OBR.2.1.1>006600610< is written
> to the DB and not the 00660610 value we want.
>
> Curious which would be the best way to handle this scenario in Mirth.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/5683>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWD5WPMQOUSYMRFKIR5DWXY7QTANCNFSM6AAAAAAU6JQVJE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***
